Implement login feature

sergipastor opened the pull request.

Commit:
Create init and import modules

Commit:
Implement first version of do_login

Commit:
Implement first version of start_requests

Commit:
Merge branch 'master' into feature/login

Commit:
Create excceptions.py

Commit:
Import modules to login.py

Commit:
Change status of FormRequest to 'POST' and update meta

Commit:
Implement check_login partially

Commit:
Handle login exceptionin auth.py

Commit:
Import login exceptions to auth.py

Commit:
Obtain check_login_selector and add it in the request

Commit:
Create first version of example json

sergipastor committed:
Fix typo

asciidiego commented on segurata/enginee/enginee/login/exceptions.py:
```diff
@@ -0,0 +1,7 @@
+class LoginGeneralError(Exception):
```
```suggestion
class LoginBaseError(Exception):
```

asciidiego commented on segurata/enginee/enginee/login/exceptions.py:
```diff
@@ -0,0 +1,7 @@
+    'Segurata was able not login.'
+    pass
+
+class NotLoggedInError(LoginGeneralError):
```
```suggestion
class NotLoggedInError(LoginBaseError):
```

asciidiego commented on segurata/enginee/enginee/spiders/auth.py:
```diff
@@ -3,11 +3,14 @@
 from scrapy.linkextractors import LinkExtractor
 from enginee.user_config.parse import user_config_parse
 from enginee.user_config.exceptions import *
+from enginee.login.login import *
```
I would try not to use the `*` operator because it's less explicit. The only exception I find feasible is the `exceptions` (haha) module, because in the future there can be TONS of them. What do you think?

```suggestion
from enginee.login.login import LoginGeneralError
```

asciidiego reviewed (commented):
I still some things I would change. Let me personally mess with the code and I will tell you. But overall it looks very good!

There are many implicit assumptions that I don't know if they are the correct ones (I honestly do not know). Such as:

1. Using the meta-data in line `32` or even sending it through the meta-data in the first place? I repeat, I do not know if this is the correct approach. Anything simpler is more than welcomed in my opinion.

2. The login method seems a bit unnecessary to me. But I do not know how I would improve it and since I say _Aporta o aparta_ I have no comments in the meantime, but I do want to create a branch from this one because I think this is the right path.

3. I love the username.

_Es que nadie piensa en los tests_?

sergipastor committed:
Update segurata/enginee/enginee/spiders/auth.py

Co-Authored-By: Diego Vincent <diego.vincent.rodriguez@gmail.com>

sergipastor committed:
Update segurata/enginee/enginee/login/exceptions.py

Co-Authored-By: Diego Vincent <diego.vincent.rodriguez@gmail.com>

sergipastor committed:
Update segurata/enginee/enginee/login/exceptions.py

Co-Authored-By: Diego Vincent <diego.vincent.rodriguez@gmail.com>

sergipastor committed:
Update example_config.json

sergipastor committed:
Update exceptions.py

asciidiego committed:
Add loginform to dependencies

Co-authored-by: Sergi Pastor <sergi.pastor.rochina@gmail.com>

asciidiego committed:
Polish login workflow of the spider

We have implemented additional features and applied the changes discussed in #7:

1. Added documentation.
2. User has the ability to pass an after-logged-in URL.
3. A functional programming style has been adopted by using the `cb_kwags` argument in callbacks.

Co-authored-by: Sergi Pastor <sergio.pastor.rochina@gmail.com>
